### ISSUE TITLE
Half-baked clang warning about address of members of packed structs

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#30085]
|  |  |
| --- | --- |
| Bugzilla Link | [30737](https://llvm.org/bz30737) |
| Version | trunk |
| OS | Linux |
| Attachments | [Minimal test case with actual code from production](https://user-images.githubusercontent.com/60944935/143754565-279d58bf-7264-4573-a707-6992272c3ce7.gz) |
| Reporter | LLVM Bugzilla Contributor |
| CC | @jmolloy,@pogo59 |

## Extended Description 

I'm using clang 4.0 on Debian and getting spurious "address-of-packed-member" warnings.

The struct in question is declared with 
```cpp
  __attribute__((packed))
  __attribute__((aligned(2)))
```
and consists of three `unsigned short`s. 

So it seems the warning is triggering without regard to the actual alignment of the struct and its members and without regard to the actual storage aligment of the struct. 

```console
$ /usr/bin/clang-4.0 -v 
clang version 4.0.0-svn283054-1~exp1 (trunk)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/i686-linux-gnu/5
Found candidate GCC installation: /usr/bin/../lib/gcc/i686-linux-gnu/5.4.1
Found candidate GCC installation: /usr/bin/../lib/gcc/i686-linux-gnu/6
Found candidate GCC installation: /usr/bin/../lib/gcc/i686-linux-gnu/6.2.0
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/5
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/6
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/6.2.0
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/5
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/5.4.1
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/6.2.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5.4.1
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.2.0
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/6.2.0
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Candidate multilib: x32;@mx32
Selected multilib: .;@m64
```
